### PR TITLE
Add harvest doctor command

### PR DESCRIPTION
**Pull request by cgrinds**
* Use `gopkg.in/yaml.v3` to parse yml - better parser with comment support
* Create a `HarvestConfig` struct - biggest advantage of this struct is it allows us to move away from [StringlyTyped](http://wiki.c2.com/?StringlyTyped) code
* Add `harvest doctor` cmd which:
  * Validates config file passed via `--config`
  * Prints a sanitized version of the config file: removing comments, usernames, passwords, and auth tokens - the goal is to include the output of this for every issue
* Testcase for sanitizing and struct creation

Remaining
* Evaluate the effort to replace the current config parsing code with `HarvestConfig` struct
* Add more checks to `checkAll`
* Evaluate the effort to create structs for everything in `conf/` folder. Haven't seen anything that looks problematic from a schema point of view

The goal is to merge this into main after review and continue improving

### CODE REVIEW

File: cmd/tools/doctor/doctor.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+func sanitize(nodes []*yaml.Node) {
+	for i, node := range nodes {
+		if node == nil {
+			continue
+		}
+		if node.Kind == yaml.ScalarNode && node.ShortTag() == "!!str" {
+			// Update this list of conditionals if there are other tokens you want to sanitize
+			value := node.Value
+			if value == "username" || value == "password" ||
+				value == "grafana_api_token" || value == "token" {
```
Comment (rahulguptajss) [inline cmd/tools/doctor/doctor.go:147]: Should this be a slice?

Comment (rahulguptajss) [reply cmd/tools/doctor/doctor.go:147]: We should redact addr also. and consult host as well.

File: cmd/tools/doctor/doctor.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+// The downside of making all members pointers is accessing
+// the values requires more dereferencing - see doctor_test.go
+
+type Consul struct {
+	Host        *string   `yaml:"host,omitempty"`
+	ServiceName *string   `yaml:"service_name,omitempty"`
+	Tags        *[]string `yaml:"tags,omitempty"`
+}
+
+type Tools struct {
```
Comment (rahulguptajss) [inline cmd/tools/doctor/doctor.go:43]: i think all these structs and yml parsing shall be defined in other go file.  May be only keep harvestdoctor related methods here?

**Comment by cgrinds** [comment]
Addressed review comments and updated testcase to include addr and host

**Review by rahulguptajss — APPROVED** [review]

**Pull request merged**